src/components/molecules/NavList.tsx

```diff
@@ -13,24 +14,36 @@ const orderedLinks: (Genre | Role)[] = [
 ]
 
 export function NavList() {
+  const location = useLocation()
+
   const secondNavLinks = useMemo(
     () =>
       orderedLinks.map(linkEnum => {
         const navLink = {
           route: enumMappings[linkEnum].url,
           label: enumMappings[linkEnum].label,
         }
-        return <NavLinks navLink={navLink} key={`nav-link-${linkEnum}`} />
+        return (
+          <NavLinks
+            navLink={navLink}
+            key={`nav-link-${linkEnum}`}
+            isSelected={location.pathname === enumMappings[linkEnum].url}
+          />
+        )
```

Comment:
:art: I see this and lines 40-44 and see duplicate code. Is there a way to refactor this to skip the duplication?

That being said, it's not a lot and creating a function for this alone might be overkill.  6-of-1

Reply:
I rewrote the `secondNavLinks` and `firstNavLinks` like this utilizing a new function called `generateNavLinks`. 
I saved the re-written code in a slack message so if you like this idea better I can easily come back in and change this file. 

```
import { List } from '@mui/material'
import { useMemo } from 'react'
import { enumMappings } from '../../common/constants/constants'
import { Genre, Role, navLinks } from '../../common/constants/enums'
import { NavLinks } from './navLinks/NavLinks'
import { useLocation } from 'react-router-dom'

const orderedLinks: (Genre | Role)[] = [
  'animation',
  'documentary',
  'director_of_photography',
  'director_producer',
  'series',
]

export function NavList() {
  const location = useLocation()

  const generateNavLink = (navLink: { route: string; label: string }, isSelected: boolean, key: string) => (
    <NavLinks
      navLink={navLink}
      key={key}
      isSelected={isSelected}
    />
  )

  const secondNavLinks = useMemo(
    () =>
      orderedLinks.map(linkEnum => {
        const navLink = {
          route: enumMappings[linkEnum].url,
          label: enumMappings[linkEnum].label,
        }
        return generateNavLink(navLink, location.pathname === enumMappings[linkEnum].url, `nav-link-${linkEnum}`)
      }),
    [location.pathname]
  )

  const firstNavLinks = useMemo(
    () =>
      navLinks.map((item, index) => (
        generateNavLink(item, location.pathname === item.route, `nav-link-${item.label}-${index}`)
      )),
    [location.pathname]
  )

  return (
    <div
      style={{
        height: '100%',
        backgroundColor: 'black',
        fontWeight: 'bold',
        color: 'white',
        fontSize: 20,
        textAlign: 'right',
        overflowY: 'hidden',
        paddingLeft: 10,
        paddingRight: 10,
      }}
    >
      <List
        sx={{
          width: '100%',
          paddingBottom: '20px',
          maxWidth: 360,
        }}
        component="nav"
      >
        {firstNavLinks}
      </List>

      <List
        sx={{
          width: '100%',
          height: '100%',
          maxWidth: 360,
        }}
        component="nav"
      >
        {secondNavLinks}
      </List>
    </div>
  )
}
```